language-design.md

```diff
@@ -38,16 +79,42 @@ you know a poorly designed language? What are the symptoms?
 
 **Response**
 
+There isn't really a definitive set of rules for determining what is (or is not) a "well-designed" language.  In general you can have an idea about how well-designed a language is by how easy it is for you to use it, how many other people use it (proportional to how long it has been around/known), and if it meets some other criteria.  For example, as Steele pointed out in his article, "the sole way to win is to plan for growth" [Steele 1998]. Thus a well-designed language needs to plan for growth, both by the developers and by users.  It should also be relatively straightforward to use - the more you have to check the documentation for a single task, the more likely it is that it was poorly designed.  "Documentation matters." [Bloch 2006].  By the same token, it should follow the principle of least surprise:
 
+> Every method should do the least surprising thing it could, given its name. If a method doesn’t do what users think it will, bugs will result. [Bloch 2006]
 
----
+Additionally it should be designed to take as much of the load off of the user as possible - for example, if a large amount of boiler plate code is required to do a certain task, that boiler plate should be pulled into the language itself and only require the minimum amount of input from the user.  "Don’t make the client do anything the library could do" [Bloch 2006]. This leads into the additional point of productivity - users should be able to use the language without fighting it, and be able to complete their tasks in a productive manner.
+
+Furthermore, in the case of a DSL, it should be usable by domain experts who have little to no knowledge of programming.
```

Comment:
I disagree, but maybe it's how we define _programming_. Domain experts need to also understand the domain of programming, otherwise they will not have the deep seeded understanding needed to make coding choices.